realclientip.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+)
+
+type realClientIPParser interface {
+	GetRealClientIP(http.Header) (net.IP, error)
+}
+
+func getRealClientIPParser(headerKey string) (realClientIPParser, error) {
+	headerKey = http.CanonicalHeaderKey(headerKey)
+
+	switch headerKey {
+	case http.CanonicalHeaderKey("X-Forwarded-For"):
+		fallthrough
+	case http.CanonicalHeaderKey("X-Real-IP"):
+		fallthrough
+	case http.CanonicalHeaderKey("X-ProxyUser-IP"):
+		return &xForwardedForClientIPParser{header: headerKey}, nil
```

Comment:
I wonder if this may be better understood like this? Not a fan of double fallthrough personally, WDYT?
```suggestion
	case http.CanonicalHeaderKey("X-Forwarded-For"), http.CanonicalHeaderKey("X-Real-IP"), http.CanonicalHeaderKey("X-ProxyUser-IP"):
		return &xForwardedForClientIPParser{header: headerKey}, nil
```